unumpy/_multimethods.py

```diff
@@ -1381,7 +1381,7 @@ def diagflat(v, k=0):
 def _tri_default(N, M=None, k=0, dtype=float):
     if M is None:
         M = N
-    mask = fromfunction(lambda i, j: j <= i + k, (N, M))
+    mask = fromfunction(lambda i, j: j <= i + k, shape=(N, M))
```

Comment:
The `fromfunction` calls can be fairly slow. Maybe we should replace with `arange` and `np.less_equal`.

Reply:
The same goes for all three defaults.

Reply:
Something like this perhaps?
```python
def _tri_default(N, M=None, k=0, dtype=float):
    if M is None:
        M = N
        
    ii, jj = arange(N).reshape(-1, 1), arange(M)
    mask = jj - ii - k <= 0
    
    return array(where(mask, 1, 0), dtype=dtype)
```

Reply:
I'd replace `mask = jj - ii - k <= 0` by `mask = jj - ii <= k` but the rest is fine.

Reply:
Or even `mask = jj <= ii + k`.

Reply:
The last one is definitely more intuitive.